[WebGPU] Update WebGPU.h to upstream ToT

litherum opened the pull request:
#### 219fc036fbe9fe884fd02b5e4412517cd59e4c6e
<pre>
[WebGPU] Update WebGPU.h to upstream ToT
<a href="https://bugs.webkit.org/show_bug.cgi?id=257640">https://bugs.webkit.org/show_bug.cgi?id=257640</a>
&lt;rdar://problem/110156868&gt;

Reviewed by Mike Wyrzykowski.

This updates WebGPU.h to the revision 245130311ef771c958ba43c322aa7ef1f9edfb8f at
<a href="https://github.com/webgpu-native/webgpu-headers.">https://github.com/webgpu-native/webgpu-headers.</a>

This does a few things:
- All objects get retain/release calls. This is really exciting because it means
      can delete PAL::WebGPU::DeviceWrapper and PAL::WebGPU::SwapChainWrapper.
      I&apos;ll do that in a follow-up patch. Also, we can delete the release calls
      from WebGPUExt.h.
- The header now has places to put NS_REFINED_FOR_SWIFT. I&apos;ve already started
      creating a Swift facade which can use this at
      <a href="https://github.com/WebKit/WebKit/pull/13849">https://github.com/WebKit/WebKit/pull/13849</a>
- The header now has places to put nullability macros. I&apos;ll do this in a follow-
      up patch.
- The device lost callback has moved from being set in a method to being part of
      the device&apos;s creation descriptor. (This means it no longer can change
      during the lifetime of the device, but instead has to be constant for a
      device.) I ended up actually deleting the implementation of this, because
      the model is significantly different now; it&apos;s now a &quot;push&quot; operation from
      the GPUP to the WP, rather than a &quot;pull&quot; model. I&apos;ll reimplement this in a
      follow-up patch.
- A few enums have been tweaked.

* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::GPUDevice::lost):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUAdapterImpl.cpp:
(PAL::WebGPU::supportedFeatures):
(PAL::WebGPU::AdapterImpl::requestDevice):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.cpp:
(PAL::WebGPU::DeviceImpl::resolveDeviceLostPromise): Deleted.
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUDevice.h:
* Source/WebGPU/WebGPU/Adapter.mm:
(wgpuAdapterReference):
* Source/WebGPU/WebGPU/BindGroup.mm:
(wgpuBindGroupReference):
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(wgpuBindGroupLayoutReference):
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::Buffer::mapAsync):
(wgpuBufferReference):
* Source/WebGPU/WebGPU/CommandBuffer.mm:
(wgpuCommandBufferReference):
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(wgpuCommandEncoderReference):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(wgpuComputePassEncoderReference):
* Source/WebGPU/WebGPU/ComputePipeline.mm:
(wgpuComputePipelineReference):
* Source/WebGPU/WebGPU/Device.h:
* Source/WebGPU/WebGPU/Device.mm:
(wgpuDeviceReference):
(wgpuDevicePopErrorScope):
(wgpuDevicePopErrorScopeWithBlock):
(WebGPU::Device::setDeviceLostCallback): Deleted.
(wgpuDeviceSetDeviceLostCallback): Deleted.
(wgpuDeviceSetDeviceLostCallbackWithBlock): Deleted.
* Source/WebGPU/WebGPU/Instance.mm:
(wgpuInstanceReference):
* Source/WebGPU/WebGPU/PipelineLayout.mm:
(wgpuPipelineLayoutReference):
* Source/WebGPU/WebGPU/PresentationContext.mm:
(wgpuSurfaceReference):
(wgpuSwapChainReference):
(wgpuSwapChainRelease):
* Source/WebGPU/WebGPU/QuerySet.mm:
(wgpuQuerySetReference):
* Source/WebGPU/WebGPU/Queue.mm:
(wgpuQueueReference):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(wgpuRenderBundleReference):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(wgpuRenderBundleEncoderReference):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(wgpuRenderPassEncoderReference):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(wgpuRenderPipelineReference):
* Source/WebGPU/WebGPU/Sampler.mm:
(wgpuSamplerReference):
* Source/WebGPU/WebGPU/ShaderModule.mm:
(wgpuShaderModuleReference):
* Source/WebGPU/WebGPU/Texture.mm:
(wgpuTextureReference):
* Source/WebGPU/WebGPU/TextureView.mm:
(wgpuTextureViewReference):
* Source/WebGPU/WebGPU/WebGPU.h:
* Source/WebGPU/WebGPU/WebGPUExt.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.cpp:
(WebKit::RemoteDevice::resolveDeviceLostPromise): Deleted.
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.h:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteDeviceProxy.cpp:
(WebKit::WebGPU::RemoteDeviceProxy::resolveDeviceLostPromise): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteDeviceProxy.h:

Canonical link: <a href="https://commits.webkit.org/264836@main">https://commits.webkit.org/264836@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0f3c07f02f7d8bac0646bb35f09a4d1020733f6e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8801 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9089 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9307 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10454 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8789 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11076 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9059 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11648 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8947 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/9952 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7804 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10612 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7249 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8048 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/15550 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/8359 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8197 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11531 "10 api tests failed or timed out") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8681 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7081 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7942 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2129 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12154 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8428 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
```diff
@@ -118,13 +118,8 @@ GPUDevice::LostPromise& GPUDevice::lost()
-        auto ref = GPUDeviceLostInfo::create(PAL::WebGPU::DeviceLostInfo::create(reason, ""_s));
-        weakThis->m_lostPromise->resolve(WTFMove(ref));
-    });
+    // FIXME: Implement this.
```
would be nice to add the bugzilla link

litherum committed:
[WebGPU] Update WebGPU.h to upstream ToT
https://bugs.webkit.org/show_bug.cgi?id=257640
<rdar://problem/110156868>

Reviewed by Mike Wyrzykowski.

This updates WebGPU.h to the revision 245130311ef771c958ba43c322aa7ef1f9edfb8f at
https://github.com/webgpu-native/webgpu-headers.

This does a few things:
- All objects get retain/release calls. This is really exciting because it means
      can delete PAL::WebGPU::DeviceWrapper and PAL::WebGPU::SwapChainWrapper.
      I'll do that in a follow-up patch. Also, we can delete the release calls
      from WebGPUExt.h.
- The header now has places to put NS_REFINED_FOR_SWIFT. I've already started
      creating a Swift facade which can use this at
      https://github.com/WebKit/WebKit/pull/13849
- The header now has places to put nullability macros. I'll do this in a follow-
      up patch.
- The device lost callback has moved from being set in a method to being part of
      the device's creation descriptor. (This means it no longer can change
      during the lifetime of the device, but instead has to be constant for a
      device.) I ended up actually deleting the implementation of this, because
      the model is significantly different now; it's now a "push" operation from
      the GPUP to the WP, rather than a "pull" model. I'll reimplement this in a
      follow-up patch.
- A few enums have been tweaked.

* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::GPUDevice::lost):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUAdapterImpl.cpp:
(PAL::WebGPU::supportedFeatures):
(PAL::WebGPU::AdapterImpl::requestDevice):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.cpp:
(PAL::WebGPU::DeviceImpl::resolveDeviceLostPromise): Deleted.
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUDevice.h:
* Source/WebGPU/WebGPU/Adapter.mm:
(wgpuAdapterReference):
* Source/WebGPU/WebGPU/BindGroup.mm:
(wgpuBindGroupReference):
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(wgpuBindGroupLayoutReference):
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::Buffer::mapAsync):
(wgpuBufferReference):
* Source/WebGPU/WebGPU/CommandBuffer.mm:
(wgpuCommandBufferReference):
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(wgpuCommandEncoderReference):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(wgpuComputePassEncoderReference):
* Source/WebGPU/WebGPU/ComputePipeline.mm:
(wgpuComputePipelineReference):
* Source/WebGPU/WebGPU/Device.h:
* Source/WebGPU/WebGPU/Device.mm:
(wgpuDeviceReference):
(wgpuDevicePopErrorScope):
(wgpuDevicePopErrorScopeWithBlock):
(WebGPU::Device::setDeviceLostCallback): Deleted.
(wgpuDeviceSetDeviceLostCallback): Deleted.
(wgpuDeviceSetDeviceLostCallbackWithBlock): Deleted.
* Source/WebGPU/WebGPU/Instance.mm:
(wgpuInstanceReference):
* Source/WebGPU/WebGPU/PipelineLayout.mm:
(wgpuPipelineLayoutReference):
* Source/WebGPU/WebGPU/PresentationContext.mm:
(wgpuSurfaceReference):
(wgpuSwapChainReference):
(wgpuSwapChainRelease):
* Source/WebGPU/WebGPU/QuerySet.mm:
(wgpuQuerySetReference):
* Source/WebGPU/WebGPU/Queue.mm:
(wgpuQueueReference):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(wgpuRenderBundleReference):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(wgpuRenderBundleEncoderReference):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(wgpuRenderPassEncoderReference):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(wgpuRenderPipelineReference):
* Source/WebGPU/WebGPU/Sampler.mm:
(wgpuSamplerReference):
* Source/WebGPU/WebGPU/ShaderModule.mm:
(wgpuShaderModuleReference):
* Source/WebGPU/WebGPU/Texture.mm:
(wgpuTextureReference):
* Source/WebGPU/WebGPU/TextureView.mm:
(wgpuTextureViewReference):
* Source/WebGPU/WebGPU/WebGPU.h:
* Source/WebGPU/WebGPU/WebGPUExt.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.cpp:
(WebKit::RemoteDevice::resolveDeviceLostPromise): Deleted.
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.h:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteDeviceProxy.cpp:
(WebKit::WebGPU::RemoteDeviceProxy::resolveDeviceLostPromise): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteDeviceProxy.h:

Canonical link: https://commits.webkit.org/264836@main
